File: 9095.py
import sys
from array import array
read = lambda: sys.stdin.readline().strip()

if __name__ == '__main__':
	input_cnt = int(read())
	rst = array('l', [1,2,4])
	for i in range(input_cnt):
		num = int(read())
		if len(rst) < num:
			for j in range(len(rst), num):
				temp = rst[j-1]+rst[j-2]+rst[j-3]
				rst.append(temp)
		print(rst[num-1])

File: beakjoon/9095.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
	int inputCnt;
	scanf("%d",&inputCnt);

	vector<int> v(11,0);
	v[1] = 1;
	v[2] = 2;
	v[3] = 4;
	for(int i =4; i<11; i++)
		v[i] = v[i-1] + v[i-2] + v[i-3];
	for(int i = 0; i < inputCnt; i++)
	{
		int temp;
		scanf("%d",&temp);
		printf("%d\n", v[temp]);
	}
	return 0;
}